src/sql/mod.rs

```diff
@@ -184,20 +184,27 @@ mod tests {
         );
     }
 
+    #[test]
+    fn option_as_null() {
+        let mut sql = SqlBuilder::new("SELECT 1 FROM test WHERE a = ?");
```

Comment:
It's a possible behavior and quite logical, but it's not an "optional filter" as declared in https://github.com/ClickHouse/clickhouse.rs/issues/119. In the case of a missing filter, the whole `a = ..` should be omitted.

@slvrtrn what do you think?

Reply:
Anyway, `None` as `NULL` is very logical but not related to "optional filters" in any possible way.

Reply:
If we take "optional filter" literally, it is more of an ORM feature (and, in my opinion, does not belong in this client).

However, I think that the right interpretation in this case is just binding None as NULL, and this is correct, so we should just merge this.

Reply:
Yes, I didn't mean to suggest that "optional filters" should be provided by this library (though in theory you could do that) - more that this pattern is fairly common and also fairly easy to implement once you can bind None/Some(_)

Reply:
> it is more of an ORM feature

It's not precisely ORM (because there is no object-relation mapping), but it requires some sort of SQL generator, yep.

> more that this pattern is fairly common

Yep, I have in production custom generator:
```
fn make_predicate(params: FilterParams) -> impl Predicate<Row = SomeRow> {
    // Describe columns that will be used in filters.
    // It generates structs named according to columns.
    columns! {
        SomeRow;
        robot_id,           u32,    RobotId;
        base_asset_id,      u32,    AssetId;
        quote_asset_id,     u32,    AssetId;
    }

    let filter = |p: FieldsFilterParams| {
        robot_id.eq_any(p.robots)
            & base_asset_id.eq_any(p.base_assets)
            & quote_asset_id.eq_any(p.quote_assets)
            & (base_asset_id.eq_any(p.assets.clone()) | quote_asset_id.eq_any(p.assets))
    };

    filter(params.include) & !filter(params.exclude)
}
```
Exactly to cover filters.

Not an area of responsibility of this crate anyway.